src/platform/packages/shared/kbn-esql-ast/src/parser/cst_to_ast_converter.ts

```diff
@@ -149,21 +149,20 @@ export class CstToAstConverter {
     const setCommandCtxs = ctx.setCommand_list();
     const singleStatement = ctx.singleStatement();
 
+    let header: ast.ESQLAstSetHeaderCommand[] | undefined;
+    // Process SET instructions and create header if they exist
+    if (setCommandCtxs && setCommandCtxs.length > 0) {
+      header = this.fromSetCommands(setCommandCtxs);
+    }
+
     // Get the main query from singleStatement
     const query = this.fromSingleStatement(singleStatement);
 
     if (!query) {
-      return undefined;
+      return Builder.expression.query([], this.getParserFields(ctx), header);
```

Comment:
I don't think we have had this before (empty command list query), this will parse some queries without a single command, even without a source command (which is required). I am not against it. Just something to keep in mind, and there might be some code downstream that expects at least a single source command.

Reply:
yes this is a good point Vadim.